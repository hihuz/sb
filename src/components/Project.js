import React from "react";

function Project({ id, title, desc, techs, projectUrl }) {
  const styles = { backgroundImage: `url("${id}.jpg")` };
  return (
    <a className="project" href={`https://github.com/${projectUrl}`} style={styles}>
      <div className="project-infos">
        <h2 className="project-title">{title}</h2>
        <p className="project-desc">{desc}</p>
        <div className="project-logos">
          {techs.map((tech) => (
            <div className="project-tech" key={tech}>
              <img src={`${tech}.png`} alt={tech} />
              {tech}
            </div>
          ))}
        </div>
      </div>
    </a>
  );
}

export default Project;
