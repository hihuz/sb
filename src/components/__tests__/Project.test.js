import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Project from "../Project";

describe("Project", () => {
  test("Snapshot", () => {
    const component = shallow(<Project id="test" author="hihuz" techs={[]} />);
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });

  test("Should point to github with passed project url", () => {
    const projectUrl = "some/url";
    const component = shallow(<Project projectUrl={projectUrl} techs={[]} />);
    const expected = `https://github.com/${projectUrl}`;
    const actual = component.find("a").props().href;
    expect(actual).toEqual(expected);
  });

  test("Should display title prop in an h2 tag", () => {
    const title = "sylvain";
    const component = shallow(<Project title={title} techs={[]} />);
    const expected = title;
    const actual = component.find("h2").text();
    expect(actual).toEqual(expected);
  });

  test("Should display desc prop in a p tag", () => {
    const desc = "boo";
    const component = shallow(<Project desc={desc} techs={[]} />);
    const expected = desc;
    const actual = component.find("p").text();
    expect(actual).toEqual(expected);
  });

  test("Should render a div with an image for each tech prop", () => {
    const techs = ["foo", "bar", "baz"];
    const component = shallow(<Project techs={techs} />);
    const children = component.find(".project-tech");
    expect(children.length).toEqual(techs.length);
    techs.forEach((tech, i) => {
      const expected = tech;
      const actual = children.at(i).text();
      expect(actual).toEqual(expected);
    });
  });
});
