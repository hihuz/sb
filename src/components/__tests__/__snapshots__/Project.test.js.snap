// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Snapshot 1`] = `
<a
  className="project"
  href="https://github.com/undefined"
  style={
    {
      "backgroundImage": "url("test.jpg")",
    }
  }
>
  <div
    className="project-infos"
  >
    <h2
      className="project-title"
    />
    <p
      className="project-desc"
    />
    <div
      className="project-logos"
    />
  </div>
</a>
`;
